PolyEngine/Core/Src/BinaryBuffer.hpp

```diff
@@ -0,0 +1,23 @@
+#pragma once
+
+#include "BaseObject.hpp"
+
+namespace Poly
+{
+
+	class CORE_DLLEXPORT BinaryBuffer : public BaseObject<>
+	{
+	public:
+		BinaryBuffer(size_t);
+		~BinaryBuffer();
+
+		void* GetBegin() const { return Begin; }
+		size_t GetSize() const { return Size; }
+		void AddData(void*, size_t);
```

Comment:
This shouldn't be necessary. Remove this method. If you need extending buffer functionality use `Dynarray<char>`